xapian-letor/ranker/listnet_ranker.cc

```diff
@@ -120,12 +120,13 @@ update_parameters(vector<double> &new_parameters, const vector<double> &gradient
 }
 
 void
-ListNETRanker::train(const std::vector<Xapian::FeatureVector> & training_data) {
+ListNETRanker::train(const std::vector<vector<Xapian::FeatureVector>>&
+		     training_data) {
     LOGCALL_VOID(API, "ListNETRanker::train", training_data);
     size_t fvv_len = training_data.size();
     int feature_cnt = -1;
     if (fvv_len != 0) {
-	feature_cnt = training_data[0].get_fcount();
+	feature_cnt = training_data[0][0].get_fcount();
```

Comment:
Similar to above, what if `training_data[0].empty()`?

Reply:
fixed in a fixup commit.